server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -305,18 +313,48 @@ public IndexInput openInput(String name, IOContext context) throws IOException {
         }
     }
 
-    public void copyFrom(Directory from, String src, String dest, IOContext context, boolean useCommonSuffix) throws IOException {
+    public String getMetadataFileAndAcquireLock(long primaryTerm, long generation, String resourceID) throws IOException {
```

Comment:
this is not an atomic operation which can cause issues like caller may end up creating a lock file for metadata file which is deleted. To make it atomic we will need mechanism in `RemoteDirectory` implementation to get lock on a file and don't allow conflicting operation on that file. 

Also better way in absence of cache for metadata files present in `RemoteDirectory` could be to create prefix using the passed in `primaryTerm` and `generation` but without `uuid` suffix. Then list the files using that prefix which will limit the result set and ideally should have only 1 file